tests/integ/scala/test_dataframe_reader_suite.py

```diff
@@ -354,6 +351,22 @@ def test_read_csv(session, mode):
     assert "is out of range" in str(ex_info)
 
 
+def test_read_csv_with_default_infer_schema(session):
+    test_file_on_stage = f"@{tmp_stage_name1}/{test_file_csv}"
+
+    with pytest.raises(SnowparkDataframeReaderException):
```

Comment:
```suggestion
    with pytest.raises(SnowparkDataframeReaderException, match=".*Please specify the schema or set session.read.options.*"):
```
You could also implement Yijuns suggestion like this. In either case match is a regex string so make sure it can actually match the error message.